tests/conftest.py

```diff
@@ -6,7 +6,11 @@
 
 from hangar import Repository
 import hangar
+from hangar.records import summarize
 
+
+# TODO: IMO, we should not have this cleanup and let tests catch issues
+# if there are any
```

Comment:
Can you expand on this? 

I believe it is actually required, since without explicitly cleaning up the past instances, when a new repository is initiated with the same python process, the old state would still be left in the singleton instance dicts. This was the fix which made tests actually work in the first place. Unless https://github.com/tensorwerk/hangar-py/pull/20 fixed this behavior (I'm not sure, haven't tested it myself) then I think it's still needed? Please correct me if I'm wrong though!

Reply:
Exactly, By cleaning up the past instances, we are avoiding a chance to find whether we have that issue fixed or not. I understand it is required if it is not fixed (for other tests to run) but my point is, we need to remember to remove this fixture once we confirm that particular issue is gone.

Reply:
Ok. Linking this discussion to #24 and #21 since they will decide how this is handled in the long term.